### PR TITLE
fix(eslint-plugin): [prefer-optional-chain] collect MetaProperty type

### LINKED ISSUE #6000: Bug: [prefer-optional-chain] Crashes when parsing an `import.meta` member with both the `||` and `!` operators
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&sourceType=module&code=JYMwBAFAhMC2AOB7ATgFwHSwKaoIboFdkAbMAHzLFWQKwEowBjRAOwGdFit1jEBzCACIAElmK9BdANxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uigB3UdCaRwYAL4gNQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
if (!import.meta.url || true) console.log("Hello");
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-optional-chain": "warn",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected ESLint to be all OK.

### Actual Result

The parser crashes:

```
Oops! Something went wrong! :(

ESLint: 8.27.0

Error: Unexpected member object type: MetaProperty
Occurred while linting /workspaces/file.ts:123
Rule: "@typescript-eslint/prefer-optional-chain"
...
```

### Additional Info

_No response_

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.43.0` |
| `@typescript-eslint/parser`            | `5.43.0` |
| `TypeScript`                           | `4.8.4` |
| `ESLint`                               | `8.27.0` |
| `node`                                 | `18.11` |

### PR DESCRIPTION
**Pull request by YuseiUeno**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6000
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Fix prefer-optional-chain cannot collect MetaProperty type bug

### CODE REVIEW

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Looks great so far, thanks!

Could you please also add an `invalid` test case to make sure meta properties work for the rule reporting complaints?

File: packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts [hunk 1]
```diff
@@ -211,6 +211,7 @@ ruleTester.run('prefer-optional-chain', rule, {
     '!foo!.bar || !foo!.bar.baz;',
     '!foo!.bar!.baz || !foo!.bar!.baz!.paz;',
     '!foo.bar!.baz || !foo.bar!.baz!.paz;',
+    '!import.meta.url || true;',
```
Comment (omril1) [inline packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts:214]: Maybe add `!new.target || true;` as well, it seems to work but it's exotic/meta enough to make me anxious

Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts:214]: +1, can't hurt to add the test!

**Comment by YuseiUeno** [comment]
I'm late but I fixed it

File: packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/tests/rules/prefer-optional-chain.test.ts]: I can just add one in 🙂

**Review by JoshuaKGoldberg — APPROVED** [review]
Looks great, thanks for this @YuseiUeno! 🙌 

I added a few tests at the end - didn't want to bug you about them, since they're just small edge cases. 😄

**Comment by YuseiUeno** [comment]
Thank you for adding test cases :)

**Pull request merged**